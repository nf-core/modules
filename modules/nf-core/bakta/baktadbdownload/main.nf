process BAKTA_BAKTADBDOWNLOAD {
    label 'process_single'

    conda "bioconda::bakta=1.6.1"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/bakta:1.6.1--pyhdfd78af_0' :
        'quay.io/biocontainers/bakta:1.6.1--pyhdfd78af_0' }"

    output:
    path "db/"              , emit: db
    path "versions.yml"     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''

    """
    bakta_db \\
        download \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bakta: \$(echo \$(bakta_db --help) 2>&1 | sed 's/.*Version: //g;s/ DOI.*//g')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''

    """
    echo "bakta_db \\
        download \\
        $args"

    mkdir db

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bakta: \$(echo \$(bakta_db --help) 2>&1 | sed 's/.*Version: //g;s/ DOI.*//g')
    END_VERSIONS
    """
}
