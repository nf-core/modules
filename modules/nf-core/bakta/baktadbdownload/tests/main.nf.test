nextflow_process {

    name "Test Process BAKTA_BAKTADBDOWNLOAD"
    script "../main.nf"
    process "BAKTA_BAKTADBDOWNLOAD"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "bakta"
    tag "bakta/baktadbdownload"

    test("Bakta database download") {

        when {
            process {
                """
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert path(process.out.db.get(0)).exists() },
                { assert snapshot(process.out.versions).match() }
            )
        }

    }

    test("Bakta database download - stub") {
    
        options "-stub"

        when {
            process {
                """
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.db + 
                    process.out.versions
                    ).match() }
            )
        }

    }

}
