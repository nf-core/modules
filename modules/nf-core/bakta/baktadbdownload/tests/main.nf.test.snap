{
    "Bakta database download": {
        "content": [
            [
                "versions.yml:md5,df9b091b08a41b7d5eef95727b7eac29"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-19T11:34:41.812416438"
    },
    "Bakta database download - stub": {
        "content": [
            [
                [
                    
                ],
                "versions.yml:md5,df9b091b08a41b7d5eef95727b7eac29"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-19T11:35:01.082923401"
    }
}